### PR TITLE
MGMT-12454: Dump inventory to get sosreport from ARM machine

### PR DESCRIPTION
**Pull request by adriengentil**
Build a basic inventory that allows the gather step from the CI to build
an sos report on the ARM machine.

### CODE REVIEW

File: ansible_files/roles/equinix/export_primary_device_connection_details/templates/inventory.j2 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{# The CI expects the node that runs test-infra to be in the "primary" group -#}
+{# Other nodes are ungrouped, and will be targeted when the playbook runs on "all" nodes -#}
+
+{%- set dump_vars=["ansible_host", "ansible_user", "ansible_ssh_private_key_file"] -%}
+{% macro build_host(hostname) -%}
+    {{ hostname -}}
+    {% for var in dump_vars %}
+        {{- " " + var }}={{ hostvars[hostname][var] }}
+    {%- endfor %}
+{% endmacro -%}
```
Comment (osherdp) [inline ansible_files/roles/equinix/export_primary_device_connection_details/templates/inventory.j2:10]: a bit too dark-magicy to me
maybe doing it by repeating the items can give it a bit of a clearer structure (and let future manipulations be more straight-forward)

Comment (osherdp) [reply ansible_files/roles/equinix/export_primary_device_connection_details/templates/inventory.j2]: yes, looks great!
thank you!

**Review by osherdp — APPROVED** [review]

**Comment by adriengentil** [comment]
I need to adapt the equinix metadata step as well

**Comment by adriengentil** [comment]
/unhold
the equinix metadata issue is not linked to this PR. It will be fixed with https://github.com/openshift/release/pull/37106

**Pull request merged**